tests/Tests/500__jupyterhub/custom-image.robot

```diff
@@ -1,51 +1,135 @@
+&{IMG_SOFTWARE} =        Software1=x.y.z
+&{IMG_PACKAGES} =        elyra=2.2.4    foo-pkg=a.b.c
+# Place holder for the imagestream name of BYON notebook created for this test run
+${IMAGESTREAM_NAME}=
 
 
 *** Test Cases ***
+Verify Admin User Can Access Custom Notebook Settings
+    Set Library Search Order  SeleniumLibrary
+    Launch Dashboard    ocp_user_name=${TEST_USER.USERNAME}    ocp_user_pw=${TEST_USER.PASSWORD}
+    ...    ocp_user_auth_type=${TEST_USER.AUTH_TYPE}    dashboard_url=${ODH_DASHBOARD_URL}
+    ...    browser=${BROWSER.NAME}    browser_options=${BROWSER.OPTIONS}
+    Open Notebook Images Page
+
 Verify Custom Image Can Be Added
-    [Documentation]    Applies the YAML and Gets the ImageStream
+    [Documentation]    Imports the custom image via UI
     ...                Then loads the spawner and tries using the custom img
     [Tags]    Tier2
     ...       ODS-1208
     Apply Custom ImageStream And Check Status
     Get ImageStream Metadata And Check Name
-    Begin Web Test
+    Verify Image Name  ${IMG_NAME}
+    Verify Image Description  ${IMG_NAME}  ${IMG_DESCRIPTION}
+    Verify User  ${IMG_NAME}  ${TEST_USER.USERNAME}
+    Launch JupyterHub Spawner From Dashboard
+
+    # These keywords need to be reworked to function here
+    #${spawner_description}=  Fetch Image Tooltip Description  ${IMAGESTREAM_NAME}
+    #${spawner_packages}=  Fetch Image Tooltip Info  ${IMAGESTREAM_NAME}
+    #${spawner_software}=  Fetch Image Description Info  ${IMAGESTREAM_NAME}
+    #Should Match  ${spawner_description}  ${IMG_DESCRIPTION}
+    #Should Match  ${spawner_software}  ${IMG_SOFTWARE}
+    #Should Match  ${spawner_packages}  ${IMG_PACKAGES}
+
+    Spawn Notebook With Arguments  image=${IMAGESTREAM_NAME}  size=Default
+    [Teardown]  Custom Image Teardown
+
+Test Duplicate Image
+    [Documentation]  Test adding two images with the same name (should fail)
+    Apply Custom ImageStream And Check Status
+    Sleep  1
+    Import New Image    ${IMG_URL}    ${IMG_NAME}    ${IMG_DESCRIPTION}
+    ...    software=${IMG_SOFTWARE}
+    ...    packages=${IMG_PACKAGES}
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    Sleep  1
+    Delete Image  ${IMG_NAME}
+    Reset Image Name
+
+Test Bad Image URL
+    [Documentation]  Test adding an image with a bad repo URL (should fail)
+    ${OG_URL}=  Set Variable  ${IMG_URL}
+    ${IMG_URL}=  Set Variable  quay.io/RandomName/RandomImage:v1.2.3
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Apply Custom ImageStream And Check Status
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    ${IMG_URL}=  Set Variable  ${OG_URL}
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Reset Image Name
+    # No image created, teardown not needed
+
+Test Bad Image Import
+    [Documentation]  Import a broken image and confirm it cannot spawn
```

Comment:
hm why using "randomstring" as image URL is more acceptable than a fake URL like `quay.io/RandomName/RandomImage:v1.2.3`? I mean, i would expect the same failure (`Unable to add notebook imag`) here and the image should not be created. Am I missing anything?

Reply:
I agree with you, but this is the current behaviour. I expect it has to do with OCP itself accepting/rejecting certain strings as URLs for images in ImageStream objects, and I doubt we have much control over that. 
This will change/be fixed when the validation pipeline is implemented.

Reply:
hm, got it..what else may break an image, other than wrong url?

Reply:
nothing else as far as I know